text/0000-signals-in-lwc-templates.md

```diff
@@ -0,0 +1,196 @@
+    subscribe((newValue: any) => void): () => void { /* subscribe to changes */ }
+}
+```
+
+__NOTE:__ _We can work the details of this protocol. It could be more simple, or hopefully similar to what is out there._
+
+### Integration with Components
+
+Components should treat signals as internal reactive state mechanisms. While a signal can technically be passed down to child components, it's recommended to pass only the necessary data, typically the `.value` of the signal, to keep the child components decoupled from the parent's state management strategy.
+
+Here’s how a parent component would pass the current value of a signal to a child component, which expects a number:
+
+```javascript
+// ParentComponent.js
+import { signal } from 'some/signals';
+
+export default class ParentComponent extends LightningElement {
+    count = signal(0);
+}
+```
+
+```html
+<!-- ParentComponent's template -->
+<template>
+    <!-- Pass the value of the signal, not the signal itself -->
+    <child-component count={count.value}></child-component>
+</template>
+```
+
+The child component simply receives a number as a property and uses it as such, unaware of the signal mechanism in the parent:
+
+```javascript
+export default class ChildComponent extends LightningElement {
+    @api count;
+}
+```
+
+This ensures that child components remain agnostic of the parent's state management implementation, promoting better component encapsulation and reuse.
+
+In terms of optimizations, the compiler can detect when a signal is passed down to a child component because of the `.value` in the template interpolation, and generate code to subscribe to the signal's changes. This would allow the parent component to update the child's public property `count` without a full re-render, leading to better performance. On the receiving end, because the child component's `count` property is not a signal, but a value, it will react to changes as usual, without any special handling.
+
+### Object Reactivity
+
+Objects can be wrapped in Signals to make them reactive. However, changes to nested properties must be handled by replacing the entire object:
+
+```javascript
+import { signal } from 'signals';
+
+export default class ExampleComponent extends LightningElement {
+    position = signal({ x: 0, y: 0 });
+
+    move(x, y) {
+        this.position.value = { ...this.position.value, x, y };
+    }
+}
+```
+
+### Connecting to External Data Stores
+
+Components can integrate with external stores by using a pattern similar to the current `@wire`` mechanism:
+
+```javascript
+import { record } from 'app/stores';
+
+export default class ExampleComponent extends LightningElement {
+    currentRecord = record;
+}
+```
+
+```html
+<template>
+    <p>{currentRecord.value.Name}</p>
+</template>
+```
+
+__Note:__ if the currentRecord's value is not available yet, it might require some additional logic to handle that case. This is similar to what developers have to do today when using `@wire`._
```

Comment:
```suggestion
__Note:__ if the currentRecord's value is not available yet, it might require some additional logic to handle that case. This is similar to what developers have to do today when using `@wire`.
```